api/test/tree-shaking/tree-shaking.test.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as assert from 'assert';
+import * as webpack from 'webpack';
+import * as path from 'path';
+import { Union } from 'unionfs';
+import { fs as mfs } from 'memfs';
+import * as realFs from 'fs';
+
+/**
+ * Verify that tree-shaking can be properly applied on the @opentelemetry/api package.
```

Comment:
Might be worth adding some additional comments here, stating that this is validating that "optional" (ie. metrics) are not included unless referenced.

Reply:
But, nice test 😄